gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua

```diff
@@ -154,710 +208,348 @@ function SWEP:GatherRagdollSample(ent)
 			end
-		end
-
-		dpanel.Think = function(s)
-			if not IsValid(bar) or not IsValid(scan) or not tester then return end
-
-			local charge = tester:GetCharge()
-
-			bar:SetValue(math.min(MAX_CHARGE, charge))
-
-			if charge < MAX_CHARGE then
-				bar:SetColor(COLOR_RED)
-
-				state:SetText(T("dna_menu_charge"))
-				state:SizeToContents()
-
-				scan:SetEnabled(false)
+		else
+			if self.ScanSuccess == 1 then
+				draw.FilteredTexture(192, 192, 128, 128, dna_screen_success, 255, screen_fontcolor)
+			elseif self.ScanSuccess == 2 then
+				draw.AdvancedText(TryT("dna_screen_match"), "DNAScannerDistanceFont", 256, 256, screen_fontcolor, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER , false, 2.5)
 			else
-				bar:SetColor(COLOR_GREEN)
-
-				if IsValid(ilist) and IsValid(ilist.SelectedPanel) then
-					scan:SetEnabled(true)
-
-					state:SetText(T("dna_menu_ready"))
-					state:SizeToContents()
-				else
-					state:SetText(T("dna_menu_select"))
-					state:SizeToContents()
-					scan:SetEnabled(false)
-				end
+				draw.FilteredTexture(192, 192, 128, 128, dna_screen_fail, 255, screen_fontcolor)
 			end
 		end
 
-		return dpanel
-	end
-
-	local printspanel = nil
-
-	local function RecvPrints()
-		local should_open = net.ReadBit() == 1
-		local num = net.ReadUInt(8)
-		local item_prints = {}
+		cam.End2D()
 
-		for i = 1, num do
-			item_prints[i] = net.ReadString()
-		end
-
-		if should_open then
-			if IsValid(printspanel) then
-				printspanel:Remove()
-			end
+		render.PopRenderTarget()
+	end
 
-			printspanel = ShowPrintsPopup(item_prints, GetTester(LocalPlayer()))
-		else
-			if not IsValid(printspanel) then return end
+	function SWEP:PreDrawViewModel()
+		self:FillScannerScreen()
+		self.Owner:GetViewModel():SetSubMaterial(0, "!scanner_screen_mat")
+	end
 
-			printspanel:UpdatePrints(item_prints)
-		end
+	function SWEP:PostDrawViewModel()
+		self.Owner:GetViewModel():SetSubMaterial(0, nil)
 	end
-	net.Receive("TTT_ShowPrints", RecvPrints)
 
-	local beep_success = Sound("buttons/blip2.wav")
+	function SWEP:DrawWorldModel()
+		self:FillScannerScreen()
+		self:DrawModel()
+	end
 
-	local function RecvScan()
-		local clear = net.ReadBit() == 1
+	function SWEP:RadarScan()
+		local target = self.ItemSamples[self.ActiveSample]
 
-		if clear then
+		if not IsValid(target) or not GetGlobalBool("ttt2_dna_radar") then
 			RADAR.samples = {}
 			RADAR.samples_count = 0
+			self.RadarPos = nil
 
 			return
 		end
 
-		local target_pos = net.ReadVector()
-
-		if not target_pos then return end
-
-		RADAR.samples = {{pos = target_pos}}
+		self.RadarPos = target:LocalToWorld(target:OBBCenter())
+		RADAR.samples = {{pos = self.RadarPos}}
 		RADAR.samples_count = 1
-
-		surface.PlaySound(beep_success)
 	end
-	net.Receive("TTT_ScanResult", RecvScan)
-
-	function SWEP:ClosePrintsPanel()
-		if not IsValid(printspanel) then return end
 
-		printspanel:Close()
-	end
+	-- target ID function
+	hook.Add("TTTRenderEntityInfo", "TTT2DNAScannerInfo", function(tData)
```

Comment:
This should be located in the function itself, instead of a hook, to be predictable etc.

Reply:
In which function exactly? Isn't `TTTRenderEntityInfo` a gamemode function?